packages/itwinui-react/src/core/utils/components/VirtualScroll.tsx

```diff
@@ -3,12 +3,16 @@
  * See LICENSE.md in the project root for license terms and full copyright notice.
  *--------------------------------------------------------------------------------------------*/
 import React from 'react';
+import * as ReactDOM from 'react-dom';
 import {
   mergeRefs,
   useResizeObserver,
   useIsomorphicLayoutEffect,
 } from '../hooks';
 
+const unstable_batchedUpdates =
+  ReactDOM.unstable_batchedUpdates ?? ((cb: () => void) => void cb());
```

Comment:
So if I understand this correctly, if we're in React 17 and there is no `ReactDOM.unstable_batchedUpdates`, then we use a callback function that executes statements immediately instead - in this case updating the VirtualScroll height. Then what was originally React 17 behavior that caused the update to not work as expected?

Reply:
no, this is for future React versions which might not have `ReactDOM.unstable_batchedUpdates`.
![](https://user-images.githubusercontent.com/9084735/217595327-e3a2d3a4-8426-458a-aaba-3c477d030d76.png)

we want our library to continue to work with future react versions, so this code is basically replacing it with a function that executes the callback immediately.

in react 17, it will still use `ReactDOM.unstable_batchedUpdates`, which is why it fixes the bug.

Reply:
I see. Would not having a fallback function at all would also work because React 18 and beyond do batching automatically? I.e use `unstable_batchedUpdates` if it exists, otherwise don't use anything

Reply:
> use unstable_batchedUpdates if it exists, otherwise don't use anything

that's what the fallback is doing though. if you're suggesting something else, maybe you can show some pseudo-code of how i would do it without the fallback

Reply:
No yea that's right. I was just a bit confused about that fallback function but it essentially does nothing. Anyway I have also manually tested this change and it works.